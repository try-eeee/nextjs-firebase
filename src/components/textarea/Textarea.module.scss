.textarea {
  width: 100%;
  height: 300pt;
  padding: 0.75rem;
  font-size: 1rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  resize: none;

  &:disabled {
    background-color: #f5f5f5;
    color: #cacaca;
    cursor: not-allowed;
  }
}
