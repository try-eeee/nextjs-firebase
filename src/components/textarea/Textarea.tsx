import clsx from "clsx";
import React from "react";
import { UseFormRegisterReturn } from "react-hook-form";
import styles from "./Textarea.module.scss";

interface TextareaProps extends React.ComponentPropsWithoutRef<"textarea"> {
  register?: UseFormRegisterReturn;
}

export function Textarea(props: TextareaProps) {
  const { register, ...rest } = props;

  return <textarea {...register} {...rest} className={clsx(styles.textarea)} />;
}
