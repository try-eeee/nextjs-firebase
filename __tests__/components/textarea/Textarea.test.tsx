import React from "react";
import { render } from "@testing-library/react";
import { Textarea } from "@/components/textarea/Textarea";

describe("Textarea UnitTest", () => {

  test("Snapshot Test Of Textarea", () => {
    const { asFragment } = render(<Textarea />);
    expect(asFragment()).toMatchSnapshot();
  });
});
