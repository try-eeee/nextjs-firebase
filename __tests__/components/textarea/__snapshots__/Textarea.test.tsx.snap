// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea UnitTest Snapshot Test Of Textarea 1`] = `
<DocumentFragment>
  <textarea
    class="textarea"
  />
</DocumentFragment>
`;
